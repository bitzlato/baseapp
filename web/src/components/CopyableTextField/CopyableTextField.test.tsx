import { shallow } from 'enzyme';
import { TestComponentWrapper } from 'src/lib/test';
import { CopyableTextField, CopyableTextFieldProps } from './';

const defaultProps: CopyableTextFieldProps = {
  value: '1F1tAaz5x1HUXrCNLbtMDqcw6o5GNn4dfE',
  fieldId: 'copy_id',
};

const setup = (props: Partial<CopyableTextFieldProps> = {}) =>
  shallow(
    <TestComponentWrapper>
      <CopyableTextField {...{ ...defaultProps, ...props }} />
    </TestComponentWrapper>,
  );

describe('CopyableTextField', () => {
  it('should render', () => {
    const wrapper = setup().render();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render 1 input tag', () => {
    const wrapper = setup().render();
    const input = wrapper.find('.custom-input');
    expect(input.length).toBe(1);
  });
});
