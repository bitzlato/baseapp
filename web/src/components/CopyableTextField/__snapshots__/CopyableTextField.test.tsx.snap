// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyableTextField should render 1`] = `
<div
  class="cr-copyable-text-field"
>
  <div
    class="input-group"
  >
    <div
      class="custom-input"
    >
      <label />
      <input
        class="form-control"
        id="copy_id"
        placeholder=""
        readonly=""
        type="text"
        value="1F1tAaz5x1HUXrCNLbtMDqcw6o5GNn4dfE"
      />
    </div>
    <div
      class="input-group-append"
    >
      <button
        class="cr-percentage-button iconButton"
        title="Copy"
        type="button"
      >
        <svg
          height="24"
          viewBox="0 0 20 20"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7 9a2 2 0 012-2h6a2 2 0 012 2v6a2 2 0 01-2 2H9a2 2 0 01-2-2V9z"
          />
          <path
            d="M5 3a2 2 0 00-2 2v6a2 2 0 002 2V5h8a2 2 0 00-2-2H5z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
